src/ansys_sphinx_theme/search/fuse_search.py

```diff
@@ -0,0 +1,112 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Module to fuse search."""
+
+import json
+import re
+
+from docutils import nodes
+
+
+class GetSearchIndex:
+    """Class to get search index."""
+
+    def __init__(self, doc_name, app):
+        """Initialize the class.
+
+        Parameters
+        ----------
+        doc_name : str
+            Document name.
+        app : Sphinx
+            Sphinx application.
+        """
+        self._doc_name = doc_name
+        self.doc_path = self._doc_name + ".html"
+        self.doc_title = app.env.titles[self._doc_name].astext()
+        self._doc_tree = app.env.get_doctree(self._doc_name)
+        self.sections = []
+
+    def _title_to_anchor(self, title: str) -> str:
+        """Convert title to anchor."""
+        return re.sub(r"[^\w\s-]", "", title.lower().strip().replace(" ", "-"))
+
+    def iterate_through_docs(self):
+        """Iterate through the document."""
+        for node in self._doc_tree.traverse(nodes.section):
+            self.section_title = node[0].astext()
+            self.section_text = "\n".join(
+                n.astext()
+                for node_type in [nodes.paragraph, nodes.literal_block]
+                for n in node.traverse(node_type)
+            )
+            self.section_anchor_id = self._title_to_anchor(self.section_title)
+            self.sections.append(
+                {
+                    "section_title": self.section_title,
+                    "section_text": self.section_text,
+                    "section_anchor_id": self.section_anchor_id,
+                }
+            )
+
+    def get_search_index(self):
+        """Get search index."""
+        search_index_list = []
+        for sections in self.sections:
+            search_index = {
+                "objectID": self._doc_name,
+                "href": f"{self.doc_path}#{sections['section_anchor_id']}",
+                "title": self.doc_title,
+                "section": sections["section_title"],
+                "text": sections["section_text"],
+            }
+            search_index_list.append(search_index)
+        return search_index_list
```

Comment:
Maybe, we could return a generator. This way, we remove allocating a list. Users can always convert:

```suggestion
    def get_search_index(self):
        """Get search index."""
        for sections in self.sections:
            search_index = {
                "objectID": self._doc_name,
                "href": f"{self.doc_path}#{sections['section_anchor_id']}",
                "title": self.doc_title,
                "section": sections["section_title"],
                "text": sections["section_text"],
            }
            yield search_index
```